src/hotspot/share/gc/g1/g1FreeIdSet.cpp

```diff
@@ -43,7 +44,7 @@ G1FreeIdSet::G1FreeIdSet(uint start, uint size) :
          "start (%u) + size (%u) overflow: ", start, size);
   // 2^shift must be greater than size. Equal is not permitted, because
   // size is the "end of list" value, and can be the index part of _head.
-  uint shift = log2_intptr((uintptr_t)size) + 1;
+  uint shift = ilog2(size) + 1;
```

Comment:
It's not obvious that it's OK to change the type here. Could you change back so that we don't have to figure that out in this review?

Reply:
The type of `size` is manifestly `uint`.  The cast doesn't do anything useful, and I have no idea why it's there, even though it looks like I wrote this code.  It certainly seems out of character for me.  And I wonder why I didn't use `log2_uint`?  Maybe that was added later?

Reply:
I agree. This seems fine.